hbase-server/src/main/java/org/apache/hadoop/hbase/master/snapshot/TakeSnapshotHandler.java

```diff
@@ -147,12 +159,16 @@ private TableDescriptor loadTableDescriptor() throws IOException {
   public TakeSnapshotHandler prepare() throws Exception {
     super.prepare();
     // after this, you should ensure to release this lock in case of exceptions
-    this.tableLock.acquire();
-    try {
-      this.htd = loadTableDescriptor(); // check that .tableinfo is present
-    } catch (Exception e) {
-      this.tableLock.release();
-      throw e;
+    if (this.tableLock.tryAcquire(this.lockAcquireTimeoutMs)) {
```

Comment:
Maybe we could use rpc timeout(maybe rpcTimeout/2?) here if this method is called through rpc?

Reply:
Maybe the default timeout could be calculated by rpcTimeout. Anyway, not a blocker issue.

Reply:
Done